omod-common/src/main/java/org/openmrs/module/webservices/rest/web/ConversionUtil.java

```diff
@@ -296,11 +297,37 @@ public static Object convert(Object object, Type toType) throws ConversionExcept
 	 */
 	@SuppressWarnings({ "rawtypes", "unchecked" })
 	public static Object convertMap(Map<String, ?> map, Class<?> toClass) throws ConversionException {
-		// TODO handle refs by fetching the object at their URI
-		Converter converter = getConverter(toClass);
-		
+		Converter converter = null;
 		Object ret = null;
 		Object uuid = map.get(RestConstants.PROPERTY_UUID);
+		
+		if (toClass.equals(String.class)) {//is referenced object, retrieve by URI
+			String uri = getSelfLink(map);
+			//TODO RestConstants.URI_PREFIX may be poiting to another server?
+			String resourceCode = uri.split("/ws/rest/v1/")[1].replace("/" + uuid.toString(), "");
+			OpenmrsObject o = null;
+			if ("concept".equals(resourceCode)) {
```

Comment:
I think this needs to be a more reliable in order to work for all resources.

I would also think that the correct thing that needs to happen here is to fetch this resource from the remote server because the referenced instance won't always exist on the machine where this is getting executed.

Reply:
I'm not sure if we want to do this though, I suggest we leave this one out in the meantime because it's very likely to be tricky to implement.

Reply:
In fact, this isn't the job of the rest module, if a property value is a reference i.e. a resource URL, the client should resolve that before it submits the data, it means this logic belongs in the client which in your use case will be the SYNC2 module, makes sense?

Reply:
Think about, it's very possible that the referenced resource doesn't exist on this server even though it does exist on the one where it was fetched from, it means it has to be fetched from the source and not this instance like you are doing.